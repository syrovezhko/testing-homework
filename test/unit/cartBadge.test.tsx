import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { createStore } from "redux";
import * as reduxHooks from "react-redux";
import { CartBadge } from "../../src/client/components/CartBadge";

const initialState = {
  cart: {},
};
function reducer(state = initialState, action: any) {
  return state;
}

const store = createStore(reducer);

describe("CartBage", () => {
  it("not item in cart store", async () => {
    const initialStoreCart = {};
    jest.spyOn(reduxHooks, "useSelector").mockReturnValue(initialStoreCart);
    const app = <CartBadge id={0} />;
    const { container } = render(app);
    expect(container.getElementsByClassName("CartBadge").length).toEqual(0);
    expect(container).toMatchSnapshot();
  });

  it("item in cart store", async () => {
    const initialStoreCart = {
      "0": {
        name: "Awesome Ball",
        count: 1,
        price: 190,
      },
    };
    jest.spyOn(reduxHooks, "useSelector").mockReturnValue(initialStoreCart);

    const app = <CartBadge id={0} />;
    const { container } = render(app);
    expect(container.getElementsByClassName("CartBadge").length).toEqual(1);
    expect(container).toMatchSnapshot();
  });
  it("items in cart store", async () => {
    const initialStoreCart = {
        "0": {
            "name": "Awesome Ball",
            "count": 1,
            "price": 190
        },
        "1": {
            "name": "Sleek Chips",
            "count": 1,
            "price": 606
        },
        "2": {
            "name": "Incredible Hat",
            "count": 1,
            "price": 245
        },
        "3": {
            "name": "Incredible Hat",
            "count": 1,
            "price": 519
        },
        "4": {
            "name": "Gorgeous Keyboard",
            "count": 1,
            "price": 300
        },
        "5": {
            "name": "Gorgeous Ball",
            "count": 1,
            "price": 762
        }
    }
    jest.spyOn(reduxHooks, "useSelector").mockReturnValue(initialStoreCart);

    const app = <CartBadge id={0} />;
    const { container } = render(app);
    expect(container.getElementsByClassName("CartBadge").length).toEqual(1);
    expect(container).toMatchSnapshot();
  });
  it("items in cart store", async () => {
    const initialStoreCart = {
        "0": {
            "name": "Awesome Ball",
            "count": 1,
            "price": 190
        },
        "1": {
            "name": "Sleek Chips",
            "count": 1,
            "price": 606
        },
        "2": {
            "name": "Incredible Hat",
            "count": 1,
            "price": 245
        },
        "3": {
            "name": "Incredible Hat",
            "count": 1,
            "price": 519
        },
        "4": {
            "name": "Gorgeous Keyboard",
            "count": 1,
            "price": 300
        },
        "5": {
            "name": "Gorgeous Ball",
            "count": 1,
            "price": 762
        }
    }
    jest.spyOn(reduxHooks, "useSelector").mockReturnValue(initialStoreCart);

    const app = <CartBadge id={10} />;
    const { container } = render(app);
    expect(container.getElementsByClassName("CartBadge").length).toEqual(0);
    expect(container).toMatchSnapshot();
  });
});