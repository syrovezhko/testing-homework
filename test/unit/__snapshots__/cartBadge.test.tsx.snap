// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartBage item in cart store 1`] = `
<div>
  <span
    class="CartBadge text-success mx-3"
  >
    Item in cart
  </span>
</div>
`;

exports[`CartBage items in cart store 1`] = `
<div>
  <span
    class="CartBadge text-success mx-3"
  >
    Item in cart
  </span>
</div>
`;

exports[`CartBage items in cart store 2`] = `<div />`;

exports[`CartBage not item in cart store 1`] = `<div />`;
